core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -42,38 +48,124 @@ angular.module('oppia').controller('RteHelperModalController', [
       $uibModalInstance.dismiss('cancel');
     };
 
+    $scope.disableSaveButtonForMathRte = function() {
+      // This method disables the save button when the Math SVG has not yet
+      // been generated but being processed.
+      if (!$scope.currentRteIsMathExpressionEditor) {
+        return false;
+      } else {
+        return (
+          $scope.tmpCustomizationArgs[0].value.
+            mathExpressionSvgIsBeingProcessed);
+      }
+    };
     $scope.save = function() {
       $scope.$broadcast('externalSave');
 
       var customizationArgsDict = {};
-      for (var i = 0; i < $scope.tmpCustomizationArgs.length; i++) {
-        var caName = $scope.tmpCustomizationArgs[i].name;
-        if (caName === 'video_id') {
-          var temp = $scope.tmpCustomizationArgs[i].value;
-          customizationArgsDict[caName] = (
-            extractVideoIdFromVideoUrl(temp.toString()));
-        } else {
-          customizationArgsDict[caName] = (
-            $scope.tmpCustomizationArgs[i].value);
+      // For the case of the math rich text components, we need to handle the
+      // saving of the generated SVG file here because the process of saving
+      // the SVG is asynchronous and the saving of SVG to the backend is to
+      // be done only after the user clicks on the save button.
+      // The saving of SVGs to the backend cannot be done in the math RTE editor
+      // because the control is passed to this function as soon as the user
+      // clicks on the save button.
+      if ($scope.currentRteIsMathExpressionEditor) {
+        // The tmpCustomizationArgs is guranteed to have only one element for
+        // the case of math rich text component.
+        var svgFile = $scope.tmpCustomizationArgs[0].value.svgFile;
+        var svgFileName = $scope.tmpCustomizationArgs[0].value.svg_filename;
+        var rawLatex = $scope.tmpCustomizationArgs[0].value.raw_latex;
+        if (rawLatex === '' || svgFileName === '') {
+          AlertsService.addWarning(
+            'The rawLatex or svgFileName for a Math expression should not ' +
+            'be empty.');
+          $uibModalInstance.dismiss('cancel');
+          return;
+        }
+        if (
+          ContextService.getImageSaveDestination() ===
+          IMAGE_SAVE_DESTINATION_LOCAL_STORAGE) {
+          ImageLocalStorageService.saveImage(svgFileName, svgFile);
+          var mathContentDict = {
+            raw_latex: $scope.tmpCustomizationArgs[0].value.raw_latex,
+            svg_filename: svgFileName
+          };
+          var caName = $scope.tmpCustomizationArgs[0].name;
+          customizationArgsDict[caName] = mathContentDict;
+          $uibModalInstance.close(customizationArgsDict);
+          $scope.currentRteIsMathExpressionEditor = false;
+          return;
+        }
+        var resampledFile = (
+          ImageUploadHelperService.convertImageDataToImageFile(svgFile));
+        AssetsBackendApiService.saveMathExpresionImage(
```

Comment:
It seems this is being saved regardless of whether `ContextService.getImageSaveDestination()` is local or not, isn't that wrong?

Reply:
No, if the `ContextService.getImageSaveDestination` gives  localStorge, then this part is never reached because there is a return statement on line 134,

Reply:
Ah, sorry, missed that!